File: admin/src/main/java/com/project/admin/domain/entity/TechBlog.java
package com.project.admin.domain.entity;

import com.project.admin.constant.Region;
import com.project.admin.constant.Source;
import com.project.admin.constant.Status;
import jakarta.persistence.*;
import java.util.List;
import lombok.Builder;
import lombok.Getter;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

@Entity
@Getter
@Table(name = "tech_blog",
uniqueConstraints = {
        @UniqueConstraint(columnNames = {"url"})
})
@NoArgsConstructor
public class TechBlog {

    @Id
    private Long id;
    @Column(name = "source")
    @Enumerated(EnumType.STRING)
    private Source source;
    @Column(name = "region")
    @Enumerated(EnumType.STRING)
    private Region region;
    @Column(name = "title")
    private String title;
    @Column(name = "url", length = 500)
    private String url;
    @Column(name = "published_at")
    private LocalDateTime publishedAt;
    @Column(name = "status")
    @Enumerated(EnumType.STRING)
    private Status status;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "tags", columnDefinition = "TEXT")
    private List<String> tags;


    @Builder
    public TechBlog(Long id, String url, Status status,LocalDateTime publishedAt) {
        this.id=id;
        this.url=url;
        this.status=status;
        this.publishedAt=publishedAt;
    }

}

File: api/src/main/kotlin/com/project/api/entity/TechBlog.kt
package com.project.api.entity

import com.project.api.constants.Region
import com.project.api.constants.Source
import com.project.api.constants.Status
import jakarta.persistence.*
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.type.SqlTypes
import java.time.LocalDateTime

@Entity
@Table(name = "tech_blog")
class TechBlog(

    @Id
    val id: Long,

    @Enumerated(EnumType.STRING)
    @Column(name = "source")
    val source: Source,

    @Enumerated(EnumType.STRING)
    @Column(name = "region")
    val region: Region,

    @Column(name = "title")
    val title: String,

    @Column(name = "url", length = 500)
    val url: String,

    @Column(name = "published_at")
    val publishedAt: LocalDateTime,

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    val status: Status,

    @OneToOne(mappedBy = "techBlog",fetch = FetchType.LAZY)
    val approval: TechBlogApproval? =null,

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "tags", columnDefinition = "json")
    val tags: List<String>?,
)
